src/python/pants/backend/python/targets/python_target.py

```diff
@@ -64,6 +74,19 @@ def __init__(
 
         self._provides = provides
 
+        # Check that the requirement constraints are well-formed.
```

Comment:
Why check this? Seems like overkill, considering the constraints aren't even supported.  Might as well just silently ignore the field entirely.